device/src/mac.rs

```diff
@@ -6,20 +6,24 @@ During Uplink assembly, this struct will be inquired to drive construction
 use heapless::Vec;
 
 use super::region;
-use lorawan_encoding::maccommands::{LinkADRAnsPayload, MacCommand};
+use lorawan_encoding::maccommands::{
+    LinkADRAnsPayload, MacCommand, RXTimingSetupAnsPayload, RXTimingSetupReqPayload,
+};
 
 #[derive(Default, Debug)]
 pub struct Mac {
     adr_ans: AdrAns,
+    rx_delay_ans: RxDelayAns,
 }
 
 type AdrAns = u8;
+type RxDelayAns = bool;
 
 //work around for E0390
 trait AdrAnsTrait {
```

Comment:
does that name still reflect well the way it's used?

Reply:
I added some comments on the Trait impl which hopefully clarifies